Several improvements to "First 3D Game" tutorial

shane-smith opened the pull request:
The tutorial has been marked as compatible with the latest version - I just ran through it and verified that each step can be completed in Godot 4.1.3 stable.

A few fixes have been made, and other sections clarified to help first-time users of Godot.

I have made no changes to images - only text.

----

Note that this also fixes the "Animation > Copy" issue mentioned in https://github.com/godotengine/godot-docs/issues/6744

- The fix is located in Step 9, starting around Line 264
- The solution I've added to the documentation is the one suggested by @riidom here: https://github.com/godotengine/godot/issues/60848#issuecomment-1670398145
- If this PR is merged, it can be marked as done in the checklist on https://github.com/godotengine/godot-docs/issues/6744#issue-1574322830

<!--
Please target the `master` branch in priority.
PRs can target other branches (e.g. `3.2`, `3.5`) if the same change was done in `master`, or is not relevant there.
PRs must not target `stable`, as that branch is updated manually.

The type of content accepted into the documentation is explained here:
https://docs.godotengine.org/en/latest/community/contributing/content_guidelines.html
-->

Calinou committed:
Add support for user-provided comments with Giscus

This allows users to leave comments on pages that don't have
`:allow_comments: False` somewhere in the page's source.
Both manual and class reference pages can receive comments.
Index pages cannot have comments, as discussion should occur on "leaf" pages.

GitHub Discussions is used as a backend on the same repository. This means
that Discussions *must* be enabled on godotengine/godot-docs before this
commit is merged to `master`. Users can choose to use the "Custom" watch
mode if they don't want to get notifications for discussion updates,
but still get notifications for issue and pull request updates.

User comments are intended to be used for the following purposes:

- Add a clarification or correct something in the documentation,
  without having to open a pull request. Contributors are encouraged to
  take a look at discussions from time to time, and see if there's information
  worth incorporating in the pages themselves. Don't forget to reply to
  the comment when doing so :)
- Mention a workaround for a common issue.
- Link to useful third-party resources that are relevant to the current page,
  such as tutorials or add-ons.

User comments should *not* be used for technical support. Other community
platforms should be used for that.

Page-to-discussion matching is done using the `pagename` Sphinx variable,
which is independent of the Godot version and documentation language.
Being independent of the Godot version allows keeping old comments
when the Godot version changes, while also allowing users from `/stable`
and `/4.1` to "see" each other in discussions.

See https://giscus.app for more information.

AThousandShips committed:
Use `Basis.look_at` for player orientation

shane-smith committed:
Several improvements to "First 3D Game" tutorial

This fixes the "Animation > Copy" issue mentioned in
https://github.com/godotengine/godot-docs/issues/6744
(see Step 09, around Line 264)

The same issue as above was referenced in
https://github.com/godotengine/godot/issues/60848
and the fix was to use the solution by @riidom

Also, a reference to a button icon has been changed
from "A+" to "A>" per feedback on PR 8460

Calinou commented on getting_started/first_3d_game/01.game_setup.rst:
```diff
@@ -29,6 +27,9 @@ Click *Import & Edit* to open the project in the editor.
 
 .. image:: img/01.game_setup/03.import_and_edit.png
 
+If you are prompted to convert the project, click *Convert Full Project*,
+and then *OK*. This will convert the project to the latest Godot version.
```
The project files should be updated for Godot 4, so this shouldn't be required.

You may see a notice because you're upgrading the project's *minor* version, but the dialog you see is different in this case:

![image](https://github.com/godotengine/godot-docs/assets/180032/55295c21-4d9c-4440-bc15-afe7fc024cbc)

shane-smith replied:
Oh it gave me a warning about upgrading from a v3 project file. But I'm happy to revert this.

Calinou commented on getting_started/first_3d_game/03.player_movement_code.rst:
```diff
@@ -12,6 +10,8 @@ Right-click the ``Player`` node and select *Attach Script* to add a new script t
 it. In the popup, set the *Template* to *Empty* before pressing the *Create*
 button.
 
+We set it to *Empty* because we want to write our own code for player movement.
```
```suggestion
it. In the popup, set the *Template* to *Empty* before pressing the *Create*
button. We set it to *Empty* because we want to write our own code for
player movement.
```

Calinou commented on getting_started/first_3d_game/09.adding_animations.rst:
```diff
@@ -259,16 +261,22 @@ node structure, you can copy them to different scenes.
+click OK. Click the *Paste* icon (clipboard) and it should appear in the window. Click OK
+to close the window.
+
+Next, make sure that the button with an "A+" icon (Autoplay on Load) and the looping
+arrows (Animation looping) are also turned on in the animation editor in the bottom panel.
+That's it; all monsters will now play the float animation.
```
The icon isn't "A+", it's more like "A>":

![image](https://github.com/godotengine/godot-docs/assets/180032/fcecaf60-05e6-4a60-a92f-c97c818e1208)

That said, I feel writing the icon as text isn't ideal as it can end up being confusing.

We should probably look into introducing inline icons in text in a future PR.

shane-smith replied:
Yes, I noticed this. It was written as A+ in other parts of the tutorials so I just copied that.

I can change this (and all the other occurrences that were already there) if that's the path forward.

Calinou reviewed (commented):
Should be good to merge after addressing suggestions.

shane-smith commented:
Thanks for the feedback! I'll work on this and get back to you.

shane-smith commented:
I will squash this into a single commit (and rebase on the current master branch) and then it's ready for re-review.

shane-smith commented:
It is ready for you now @Calinou - let me know if you have any further feedback 🙂

akien-mga committed:
Fix incomplete thirdparty guidelines on Complying with licenses

The three libraries which had been singled out are just a small portion of
what actually requires attribution. Listing them all would basically duplicate
all of COPYRIGHT.txt, aside from CCO/PD licensed stuff.

Clarify that this is no legal advice.

Also merge in grammar and clarity improvements by John from another PR.

Co-authored-by: John Veness <john.veness.github@pelago.org.uk>

skyace65 commented:
This PR would supersede #8197. My preference is this one.

skyace65 reviewed (approved):
This looks good, I'd prefer to leave the "article outdated" warnings for now. There's still stuff on other pages that needs to be updated or fixed. but I'd be okay with merging this as is if others think it's fine.

I'll work on a PR to address the remaining issues once this one is merged.

shane-smith commented:
I'll restore the warnings 👍

AdriaandeJongh committed:
added section on common unsafe operations and their safe counterparts

jsjtxietian committed:
Update textureQueryLevels to mention it returns 1 when the texture is unassigned

mhilbrunner committed:
Merge pull request #8664 from bruvzg/win_cmds

Fix Windows build command highlighting.

skyace65 committed:
Merge pull request #7532 from jynus/drawing

Update custom drawing in 2d tutorial to Godot4

Commit:
classref: Sync with current master branch (13a0d6e)

skyace65 committed:
Merge pull request #8624 from bruvzg/upd_d3d12_docs

Update D3D12 build instructions.

skyace65 committed:
Merge pull request #8669 from jsjtxietian/patch-1

Update textureQueryLevels to mention it returns 1 when the texture is unassigned

GreentheNinja committed:
[Minor] Fix typo "Rougness" (should be "Roughness")

Vercix committed:
Update OS to Time on pages containing OS.get_ticks_usec()

skyace65 committed:
Merge pull request #8682 from Vercix/master

Update OS to Time on pages containing OS.get_ticks_usec()

ramadm committed:
Fixed typo in pr_workflow.rst (#8671)

* Fixed typo in pr_workflow.rst

skyace65 committed:
Merge pull request #8681 from GreentheNinja/patch-1

[Minor] Fix typo in `importing_images.rst` ("Rougness" should be "Roughness")

mhilbrunner committed:
Merge pull request #8674 from godotengine/classref/sync-13a0d6e

classref: Sync with current master branch (13a0d6e)

PgBiel committed:
Update link to Rust GDExtension bindings

skyace65 committed:
Merge pull request #8687 from PgBiel/patch-1

Update link to Rust GDExtension bindings

flametime committed:
Update changed signals

Old signals in doc does not exist

brentruhle committed:
Verified step by step index is up to date

Reviewed the documentation on the main index page of step by step and have verified it is up to date. Setting article outdated flag to false.

brentruhle committed:
Adding missing MacOS commands

Singular instance of the the F5 build command not having it's corresponding MacOS shortcut equivalent.

seadylan committed:
Fixed unordered list markup in "Converting GLSL to Godot shaders"

The unordered list markup was missing a blank line at the beginning, so it displayed all the list item on a single line of text instead of in a proper bulleted list.

BenAmundson committed:
Update 05.spawning_mobs.rst (typo regarding 'respectively')

The order of mentioned nodes (SpawnPath and SpawnLocation) does not match the previous sentence when using "respectively". The following image confirms the order may be backwards.

fellowjamess committed:
Update introduction.rst

Changed Q&A to forum.

mhilbrunner committed:
Merge pull request #8691 from brentruhle/patch-1

Verified step by step index is up to date

mhilbrunner committed:
Merge pull request #8688 from flametime/patch-1

Update changed signals

mhilbrunner committed:
Merge pull request #8693 from brentruhle/patch-2

Adding missing MacOS commands

eztaah committed:
Fix incorrect instruction to call node during 2D tutorial

trflorian committed:
replace yield with await in gd script for http client tutorial, use await instead of synchronous OS delay also on non-web platforms

skyace65 committed:
Merge pull request #8702 from N3ST0R-byte/patch-2

Update introduction.rst

jsjtxietian committed:
Update spatial_shader.rst to mention godot use the same VIEW vector for both perspective and orthogonal cameras.

IntangibleMatter committed:
Fix export annotation being formatted as Godot 3.

While browsing the docs, I noticed that there was a bit of code obviously misformatted for Godot 4. Fixed it, hopefully.

eztaah committed:
Reorder instructions in 06.heads_up_display.rst

smix8 committed:
Improve navigation mesh code examples

Improves navigation mesh code examples.

skyace65 committed:
Merge pull request #8713 from eztaah/fix1

Reorder instructions in 06.heads_up_display.rst

mhilbrunner committed:
Merge pull request #8708 from jsjtxietian/patch-2

Update spatial_shader.rst to mention godot use the same VIEW vector for both perspective and orthogonal cameras.

smix8 committed:
Add script example for NavigationLink

Adds a script example how to create NavigationLinks with the NavigationServer.

akien-mga committed:
Merge pull request #8657 from aaronfranke/template-target

Update export template docs to use Godot 4.x target names

FishingSimulator committed:
Update "The Main Game Scene" for clarity

Many people have reported having issues connecting HUD's start_game and the main node's new_game() in the next step of this tutorial. This occurred for me because the documentation states that "new_game()" should be removed from the _ready() function, but does not include instructions to add "pass" back (and pass is not left in the example code when "new_game()" is added to the function, so the initial one will not remain). Having the function declaration without "pass" breaks all the following functions, preventing new_game() (or any Main functions) from being connected.

kylestarr committed:
Update scripting_player_input.rst

Commenting out code from the previous exercise that causes the icon to float up, making it difficult for users to see the example of directional input.

jsjtxietian committed:
Update lights_and_shadows.rst to mention the increased drawcall by enabling shadow

Commit:
classref: Sync with current master branch (89cc635)

nightblade9 committed:
Update TTS page to mention Text to Speech project setting

mhilbrunner committed:
Merge pull request #8727 from godotengine/classref/sync-89cc635

classref: Sync with current master branch (89cc635)

mhilbrunner committed:
Merge pull request #8723 from jsjtxietian/patch-2

Update lights_and_shadows.rst to mention the increased drawcall by enabling shadow

Marenz committed:
Update tutorials/scripting/gdscript/gdscript_basics.rst

Lambdas need an explicit `return`

Extend the documentation so that it is clear that lambdas require an explicit return .

Also add an according example.

Co-Authored-By: Danil Alexeev <danil@alexeev.xyz>

skyace65 committed:
Merge pull request #8711 from IntangibleMatter/patch-1

Fix export annotation being formatted as Godot 3.

skyace65 committed:
Merge pull request #8707 from trflorian/http-client-await

Replace yield with await in gd script for http client tutorial

a-johnston committed:
Add c# script name note to scripting_first_script.rst

Newt6611 committed:
Update internal_rendering_architecture link format

skyace65 committed:
Merge pull request #8734 from Newt6611/interanl_redering

Update internal_rendering_architecture link format

kylestarr committed:
Update getting_started/step_by_step/scripting_player_input.rst

Removing unnecessary spaces in C# comments

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

mhilbrunner committed:
Merge pull request #8728 from nightblade9/master

Update TTS page to mention Text to Speech project setting

mhilbrunner committed:
Merge pull request #8730 from Marenz/patch-1

Lambdas need an explicit `return`

mhilbrunner committed:
Merge pull request #8716 from smix8/navlink_example

Add script example for NavigationLink

mhilbrunner committed:
Merge pull request #8714 from smix8/navmesh_examples

Improve navigation mesh code examples

mhilbrunner committed:
Merge pull request #8593 from Calinou/integer-scaling-fullscreen-mode

Document Fullscreen vs Exclusive Fullscreen window mode for integer scaling

mhilbrunner committed:
Merge pull request #8654 from AdriaandeJongh/patch-3

Added section on common unsafe operations and their safe counterparts

jsjtxietian committed:
Update lights_and_shadows.rst to mention difference between light with or without shadow

bruvzg committed:
[iOS] Remove mention of Command Line Tools.

BastiaanOlij committed:
Corrected a few things in the openxr android documentation

skyace65 committed:
Merge pull request #8740 from BastiaanOlij/improve_openxr_android_export_42

Corrected a few things in the openxr android documentation

Daniel-The-Fox committed:
Add info reg. universal binaries to article on exporting for macOS

theDrake committed:
Fix typo: 'generated' -> 'generate'

Minor typo fix: "used to generated" -> "used to generate"

skyace65 committed:
Merge pull request #8747 from theDrake/patch-1

Fix typo: 'generated' -> 'generate'

skyace65 committed:
Merge pull request #8732 from a-johnston/your_first_cs_script

Add c# script name note to scripting_first_script.rst

Calinou committed:
Document importing SVG images with text

- Mention editor plugins and custom node icons should use the appropriate
  import options.

Commit:
classref: Sync with current master branch (26b1fd0)

BastiaanOlij committed:
Add a chapter to the XR manual that improves the startup script for OpenXR

mhilbrunner committed:
Merge pull request #8757 from godotengine/classref/sync-26b1fd0

classref: Sync with current master branch (26b1fd0)

mhilbrunner committed:
Merge pull request #8748 from Calinou/doc-importing-images-svg-text

Document importing SVG images with text

mhilbrunner committed:
Pin sphinxcontrib-applehelp, fix RTD Sphinx build

mhilbrunner committed:
Merge pull request #8760 from mhilbrunner/fix-rtd-sphinx-applehelp

Pin sphinxcontrib-applehelp, fix RTD Sphinx build

mhilbrunner committed:
Pin more sphinx-contrib packages

mhilbrunner committed:
Merge pull request #8761 from mhilbrunner/fix-rtd-part-2

Pin more sphinx-contrib packages

mhilbrunner committed:
Pin more sphinx-contrib packages

mhilbrunner committed:
Merge pull request #8762 from mhilbrunner/fix-rtd-part-3

Pin more sphinx-contrib packages

mhilbrunner committed:
Merge pull request #8739 from bruvzg/ios_no_cmd

[iOS] Remove mention of Command Line Tools.

mhilbrunner committed:
Merge pull request #8705 from eztaah/master

Fix incorrect instruction to call node during 2D tutorial

scotmcp committed:
Update high_level_multiplayer.rst (#8697)

* Update high_level_multiplayer.rst

Clarifying RPC checksum, and how all RPCs on a script must have matching partners on the target peer's script as well.

---------

Co-authored-by: Max Hilbrunner <mhilbrunner@users.noreply.github.com>
Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

mhilbrunner committed:
Merge pull request #8627 from godotengine/dependabot/github_actions/actions/upload-artifact-4

Bump actions/upload-artifact from 3 to 4

mhilbrunner committed:
Merge pull request #8746 from Daniel-The-Fox/Add-target-silicon-for-macos-exports

Add info reg. universal binaries to article on exporting for macOS

mhilbrunner committed:
Merge pull request #8547 from bruvzg/ios_one_click

[One click deploy] Add iOS and desktop one-click deploy info.

mhilbrunner committed:
Merge pull request #7520 from Calinou/add-page-comments

Add support for user-provided comments with Giscus

mhilbrunner committed:
Merge pull request #8607 from ShatReal/fix-using-inputevent

Areas have the input pickable property set to true as default, so I removed the part that said they didn't.

AdriaandeJongh committed:
Added xcode-select fix to iOS export page (#8631)

* Added xcode-select fix to iOS export page

---------

Co-authored-by: Max Hilbrunner <mhilbrunner@users.noreply.github.com>

Calinou committed:
Document Windows limitation to 4 controllers in Controllers, gamepads and joysticks (#8594)

* Document Windows limitation to 4 controllers in Controllers, gamepads and joysticks

---------

Co-authored-by: Max Hilbrunner <mhilbrunner@users.noreply.github.com>
Co-authored-by: Matthew <matthewehr@hotmail.com>

mhilbrunner committed:
Merge pull request #8721 from FishingSimulator/patch-1

Update "The Main Game Scene" for clarity

mhilbrunner committed:
Merge pull request #8586 from jonri/patch-1

Clarify path usage for command line options

mhilbrunner committed:
Update community channels

dalexeev committed:
GDScript: Document variable initialization order

mhilbrunner committed:
Merge pull request #8694 from seadylan/patch-1

Fixed unordered list markup in "Converting GLSL to Godot shaders"

docrinehart committed:
Replace screenshots showing animations against root Player node

docrinehart committed:
Add warning text when starting section about keyframes

docrinehart committed:
Update keyframe icon/marker language to be more generic

In Godot 4.2.1.stable, the icons are represented by squares instead of diamonds.

docrinehart commented:
I don't think there will be any conflicts between this branch and the PR I submitted, but tagging this here for tracking purposes: 

#8772

hakro committed:
Explain what the different buttons do in the Export menu

31 committed:
Update c_sharp_differences.rst ToSignal example

docrinehart committed:
Convert replaced images to webp format per formatting guidelines

Calinou committed:
Update Internal rendering architecture for Godot 4.2

skyace65 committed:
Merge pull request #8774 from 31/patch-1

Update c_sharp_differences.rst ToSignal example

skyace65 committed:
Merge pull request #8699 from BenAmundson/patch-2

Update 05.spawning_mobs.rst (typo regarding 'respectively')

QuebleGameDev committed:
Update tutorials.rst

Added my (Queble) YouTube link.
Contains general Godot tutorials.

mhilbrunner committed:
Merge pull request #8776 from Calinou/update-internal-rendering-architecture

Update Internal rendering architecture for Godot 4.2

Commit:
classref: Sync with current master branch (0bcc0e9)

sellenth committed:
improve comma placement

skyace65 committed:
Merge pull request #8804 from sellenth/patch-1

improve comma placement

31 committed:
Update compiling_with_dotnet.rst: clarify NuGet source creation

docrinehart committed:
Update images to include corrected keyframe icons

docrinehart committed:
Revert "Update keyframe icon/marker language to be more generic"

This reverts commit 81e598daa07f14ac5ea0d256cd303d1fe0606cd5.

docrinehart committed:
Remove warning comment regarding player vs character nodes

BWpanda committed:
Re-word 'Comment spacing' note in gdscript_styleguide.rst

mhilbrunner committed:
Merge pull request #8800 from godotengine/classref/sync-0bcc0e9

classref: Sync with current master branch (0bcc0e9)

31 committed:
Elaborate on C# Connect/Disconnect

31 committed:
Apply suggestions from code review

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>
Co-authored-by: Paul Joannon <437025+paulloz@users.noreply.github.com>

31 committed:
Remove issue ref; try "This means"

skyace65 committed:
Merge pull request #8773 from hakro/explain-export-buttons

Explain what the different buttons do in the Export menu

skyace65 committed:
Merge pull request #8759 from BastiaanOlij/a_better_xr_start_script

Add a chapter to the XR manual that improves the startup script.

skyace65 committed:
Merge pull request #8812 from 31/dev/31/signal-connect

Elaborate on C# Connect/Disconnect

skyace65 committed:
Merge pull request #7890 from AThousandShips/pos_fix

Fix use of global position in 3d game tutorial

skyace65 committed:
Merge pull request #8809 from BWPanda/patch-1

Re-word 'Comment spacing' note in gdscript_styleguide.rst

skyace65 committed:
Merge pull request #8805 from 31/dev/31/build-net-win-path

Update compiling_with_dotnet.rst: clarify NuGet source creation

skyace65 committed:
Update note about resources in instancing page

skyace65 committed:
Fix sub-section title style

jsjtxietian committed:
Update spatial_shader.rst to delete outdated light shader usage

hsandt committed:
Exporting for Android: fix URI fragment in hyperlink to Android SDK command line tools

Closes #8819

adamscott committed:
Remove Giscus from search page (#8777)

mhilbrunner committed:
Merge pull request #8817 from skyace65/Localization-Titles

Fix sub-section title style

mhilbrunner committed:
Merge pull request #8738 from jsjtxietian/patch-3

Mention light appearance difference between with or without shadow in compatibility

Splizard committed:
Add Go to the list of community maintained GDExtension bindings. (#8808)

* Add Go to the list of community maintained bindings.

---------

Co-authored-by: Raul Santos <raulsntos@gmail.com>

skyace65 committed:
Merge pull request #8820 from hsandt/fix-android-sdk-command-line-tools-uri-fragment

Exporting for Android: fix hyperlink URI fragment (Android command line tools)

mhilbrunner committed:
Merge pull request #8816 from skyace65/InstancingNote

Update note about resources in instancing page

mhilbrunner committed:
Merge pull request #8818 from jsjtxietian/patch-5

Update spatial_shader.rst to delete outdated light shader usage

mhilbrunner committed:
Merge pull request #8791 from QuebleGameDev/master

Added Queble YouTube link in 'tutorials.rst'

mhilbrunner committed:
Merge pull request #8766 from mhilbrunner/community-channels-update

Update community channels

AThousandShips committed:
Fix method bind instructions and style (#8735)

* Fix method bind instructions and style

---------

Co-authored-by: Max Hilbrunner <mhilbrunner@users.noreply.github.com>
Co-authored-by: Rémi Verschelde <rverschelde@gmail.com>

jsjtxietian committed:
Update thread_safe_apis.rst to mention don't call certain rendering functions on other threads (#8749)

* Update thread_safe_apis.rst to mention don't call certain rendering functions on other threads

---------

Co-authored-by: Max Hilbrunner <mhilbrunner@users.noreply.github.com>
Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

adamscott committed:
Add Seaven Studio to third-party support list

- Changed the contact link to the Godot Foundation

mhilbrunner committed:
Update community tutorials and resources for 4.0+

Co-authored-by: Radiant <69520693+RadiantUwU@users.noreply.github.com>

mhilbrunner committed:
Merge pull request #8825 from mhilbrunner/update-community-channels

Update community tutorials and resources for 4.0+

mhilbrunner committed:
Merge pull request #8824 from adamscott/seaven-studio-and-instructions-update

Add Seaven Studio to third-party support list and add Godot Foundation link and note

mhilbrunner committed:
Remove reference to outdated VS extension for C#

mhilbrunner committed:
Merge pull request #7507 from dalexeev/gds-var-init-order

GDScript: Document variable initialization order

mhilbrunner committed:
Merge pull request #7198 from Calinou/add-system-requirements

Add a page on Godot system requirements

mhilbrunner committed:
Merge pull request #8826 from mhilbrunner/remove-outdated-extension

Remove reference to outdated VS extension for C#

akien-mga committed:
Merge pull request #8513 from akien-mga/properly-complying-with-licenses

Fix incomplete thirdparty guidelines on Complying with licenses

skyace65 committed:
Update note the renderer to use for XR projects

skyace65 committed:
Document occluder import hint

mhilbrunner committed:
Merge pull request #8827 from skyace65/XR_Renderer_Note

Update note the renderer to use for XR projects

mhilbrunner committed:
Merge pull request #8828 from skyace65/OccluderHint

Document occluder import hint

Calinou committed:
Document accessing web one-click deploy from remote devices

kylestarr committed:
Update getting_started/step_by_step/scripting_player_input.rst

Updating phrasing per suggestion

Co-authored-by: Matthew <matthewehr@hotmail.com>

kylestarr committed:
Update scripting_player_input.rst

Informing user to uncomment the commented code.

Calinou committed:
Fix table caption text color on dark theme

skyace65 committed:
Re add link to learn GDScript from zero

Commit:
classref: Sync with current master branch (17e7f85)

mhilbrunner committed:
Merge pull request #8844 from godotengine/classref/sync-17e7f85

classref: Sync with current master branch (17e7f85)

AnOpenSauceDev committed:
Fix multiple typos related to Particle Systems (#8845)

* Fix typo: Ammount -> Amount

* Fix typos in `properties.rst`

Changed "Ammount" to "Amount", and "Radio" to "Ratio".

skyace65 committed:
Fix image and code in signals introduction

31 committed:
Add C# signal automatic disconnection info

31 committed:
Update external_editor.rst: point out C# instructions

31 committed:
singletons_autoload.rst: add C# example similar to "Global Variable"

skyace65 committed:
Merge pull request #8851 from 31/dev/31/external-editor-cs

Update external_editor.rst: point out C# instructions

31 committed:
c_sharp_exports.rst: more consistent public style; minor grammar

mhilbrunner committed:
Merge pull request #8843 from skyace65/Learn-GDScript

Re-add page for learn GDScript from zero

mhilbrunner committed:
Merge pull request #8848 from skyace65/SignalsIntro

Fix image and code in signals introduction

mhilbrunner committed:
Merge pull request #8840 from Calinou/theme-fix-caption-color

Fix table caption text color on dark theme

mhilbrunner committed:
Merge pull request #8829 from Calinou/one-click-deploy-web-access-remote

Document accessing web one-click deploy from remote devices

skyace65 committed:
Update images on instancing page for 4.2

skyace65 committed:
Clarify what Navlink does and does not do

31 committed:
c_sharp_exports.rst: clarify why exporting nodes/resources is useful

31 committed:
c_sharp_exports.rst: add complex getter/setter note

31 committed:
c_sharp_exports.rst: move code comments to text

DSOE1024 committed:
Update compiling_for_linuxbsd.rst

Added a description that compiling for RISC-V devices requires the use of the Clang compiler

31 committed:
Simplify node/resource export in "Basic use"

Calinou committed:
Update recommended MSVC version in Compiling for Windows

- Mention that official Godot binaries are compiled using MinGW
  (and describe why).

Calinou committed:
Disallow page comments on System requirements

Like other pages in the About section, user notes can't be posted.

skyace65 committed:
Merge pull request #8860 from 31/dev/31/export-node-res-useful

c_sharp_exports.rst: clarify why exporting nodes/resources is useful

skyace65 committed:
Merge pull request #8866 from Calinou/system-requirements-disallow-comments

Disallow page comments on System requirements

skyace65 committed:
Merge pull request #8861 from 31/dev/31/export-code-comments

c_sharp_exports.rst: move code comments to text

mhilbrunner committed:
Merge pull request #8865 from Calinou/compiling-for-windows-update-msvc

Update recommended MSVC version in Compiling for Windows

mhilbrunner committed:
Merge pull request #8854 from skyace65/Instancing-Images

Update images on instancing page for 4.2

mhilbrunner committed:
Merge pull request #8815 from 31/dev/31/signal-connect

Add C# signal automatic disconnection info

mhilbrunner committed:
Merge pull request #8853 from 31/dev/31/clean-export-cs

c_sharp_exports.rst: fix private naming style; minor grammar

scotmcp committed:
Correcting reference link

line 2016 references Class, should reference Resource

scotmcp committed:
Fixed Type description from resource to GDScript

scotmcp committed:
Update tutorials/scripting/gdscript/gdscript_basics.rst

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

dependabot[bot] committed:
Bump peter-evans/create-pull-request from 5 to 6

Bumps [peter-evans/create-pull-request](https://github.com/peter-evans/create-pull-request) from 5 to 6.
- [Release notes](https://github.com/peter-evans/create-pull-request/releases)
- [Commits](https://github.com/peter-evans/create-pull-request/compare/v5...v6)

---
updated-dependencies:
- dependency-name: peter-evans/create-pull-request
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

Commit:
Update gui_using_fonts.rst

Add link to *open-source* alternative of BMFont editor

mhilbrunner committed:
Merge pull request #8859 from 31/dev/31/backing-export-behavior

c_sharp_exports.rst: add complex getter/setter note

BWpanda committed:
Update exporting_for_android.rst

skyace65 committed:
Merge pull request #8870 from scotmcp/patch-2

Correcting link to class Reference

skyace65 committed:
Merge pull request #8772 from docrinehart/first-3d-game-animation-fixes

First 3d game animation fixes

31 committed:
c_sharp_variant.rst: add more examples, fix explicit vs. implicit

31 committed:
Fix C# typos: capitalize method name, use public property names

Synkied committed:
Update process_material_properties.rst

Calinou committed:
Fix missing texture mipmap bias formula in Resolution scaling

skyace65 committed:
Replace set_velocity with velocity property

skyace65 committed:
Merge pull request #8884 from Synkied/patch-1

Update process_material_properties.rst

skyace65 committed:
Merge pull request #8879 from 31/dev/31/my-typos

Fix C# typos: capitalize method name, use public property names

skyace65 committed:
Add instruction to bake navigation BakePolygon

Commit:
classref: Sync with current master branch (10e1114)

mhilbrunner committed:
Merge pull request #8888 from godotengine/classref/sync-10e1114

classref: Sync with current master branch (10e1114)

mhilbrunner committed:
Merge pull request #8876 from BWPanda/patch-1

Update Android SDK command line instructions

31 committed:
Changes and fixes from review

31 committed:
Apply suggestions from code review

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

31 committed:
More directly describe .Obj boxing

31 committed:
Link to C# Variant-compatible section directly

31 committed:
c_sharp_collections.rst: copy variant page's Packed{TYPE}Array order

skyace65 committed:
Merge pull request #8895 from 31/dev/31/packed-order

c_sharp_collections.rst: copy variant page's Packed{TYPE}Array order

skyace65 committed:
Remove reference to DevDocs copy of documentation

skyace65 committed:
Merge pull request #8885 from Calinou/resolution-scaling-fix-mipmap-bias-formula

Fix missing texture mipmap bias formula in Resolution scaling

skyace65 committed:
Remove outdated references to other tutorial

mhilbrunner committed:
Merge pull request #8897 from skyace65/Custom-Post-Processing

Remove outdated references to other tutorial

mhilbrunner committed:
Merge pull request #8878 from 31/dev/31/variant-obj

c_sharp_variant.rst: add more examples, fix explicit vs. implicit

mhilbrunner committed:
Merge pull request #8894 from 31/dev/31/variant-ref

Link to C# Variant-compatible section directly

mhilbrunner committed:
Merge pull request #8896 from skyace65/DevDocs

Remove reference to DevDocs copy of documentation

mhilbrunner committed:
Merge pull request #8873 from godotengine/dependabot/github_actions/peter-evans/create-pull-request-6

Bump peter-evans/create-pull-request from 5 to 6

mhilbrunner committed:
Merge pull request #8887 from skyace65/BakePolygon

Add instruction to bake navigation BakePolygon

mhilbrunner committed:
Merge pull request #8856 from skyace65/NavLinkClarity

Clarify what Navlink does and does not do

mhilbrunner committed:
Merge pull request #8886 from skyace65/VelocityChange

Replace set_velocity with velocity property

skyace65 committed:
Merge pull request #8722 from kylestarr/master

Update scripting_player_input.rst

smix8 committed:
Fix script error in NavigationMesh example

Fixes script error in NavigationMesh example.

mhilbrunner committed:
Merge pull request #8899 from smix8/navmesh_example

Fix script error in NavigationMesh example

mhilbrunner committed:
Merge pull request #8852 from 31/dev/31/singleton-cs

singletons_autoload.rst: add C# example similar to "Global Variable"

skyace65 committed:
Remove duplicate information in autoload page

adamscott committed:
Update the Emscripten minimum requirements

VojtaStruhar committed:
Typo fix

skyace65 committed:
Merge pull request #8905 from adamscott/update-emscripten-minimum-requirements

Update the Emscripten minimum requirements to 3.1.39

skyace65 committed:
Merge pull request #8907 from VojtaStruhar/fix/dedicated-servers-typo

Typo fix

skyace65 committed:
Build offline version of stable branch

skyace65 committed:
Add info on finding script templates folder

mhilbrunner committed:
Merge pull request #8910 from skyace65/ScriptTemplates

Add info on finding script templates folder

mhilbrunner committed:
Merge pull request #8909 from skyace65/BuildStable

Build offline version of stable branch

mhilbrunner committed:
Merge pull request #8875 from dustdfg/patch-1

Update gui_using_fonts.rst

dalexeev committed:
GDScript: Document `is not` operator

mhilbrunner committed:
Merge pull request #8918 from dalexeev/gds-is-not-operator

GDScript: Document `is not` operator

mhilbrunner committed:
Merge pull request #8900 from skyace65/Autoload

Remove duplicate information in autoload page

idlist committed:
Update to newer `CustomMinimumSize` API

mhilbrunner committed:
Merge pull request #8927 from idlist/patch-1

Update C# API in the code example of "Custom GUI controls"

skyace65 committed:
Clarify which scene to run in step by step

skyace65 committed:
Clarify which track is being talked about

skyace65 committed:
Remove Godot 3 reference from profiler page

AThousandShips committed:
Merge pull request #8935 from skyace65/DebugGodot3

Remove Godot 3 reference from profiler page

AThousandShips committed:
Merge pull request #8934 from skyace65/AnimationTrackClarify

Clarify which track is being talked about

AThousandShips committed:
Merge pull request #8933 from skyace65/SignalsSceneClarity

Clarify which scene to run in step by step

Commit:
classref: Sync with current master branch (4e990cd)

mhilbrunner committed:
Merge pull request #8939 from godotengine/classref/sync-4e990cd

classref: Sync with current master branch (4e990cd)

skyace65 committed:
Merge pull request #8863 from DSOE1024/patch-1

Update compiling_for_linuxbsd.rst Added a description that compiling for RISC-V devices

skyace65 committed:
Merge branch 'patch-first-3d-game' of https://github.com/shane-smith/godot-docs into patch-first-3d-game

skyace65 commented:
Salved by #8955
